django/thunderstore/community/models/community.py

```diff
@@ -206,3 +198,67 @@ def valid_review_statuses(self):
             PackageListingReviewStatus.approved,
             PackageListingReviewStatus.unreviewed,
         )
+
+
+@dataclass
+class AggregatedFields:
+    download_count: int = 0
+    package_count: int = 0
+
+
+class CommunityAggregatedFields(TimestampMixin, models.Model):
+    """
+    Computationally heavy fields updated by a periodic background task.
+    """
+
+    download_count = models.PositiveIntegerField(default=0)
+    package_count = models.PositiveIntegerField(default=0)
+
+    def __str__(self):
+        return str(self.as_class())
+
+    @classmethod
+    def get_empty(cls) -> AggregatedFields:
+        return AggregatedFields()
+
+    def as_class(self) -> AggregatedFields:
+        """
+        Return fields in a format supporting attribute access.
+        """
+        return AggregatedFields(self.download_count, self.package_count)
+
+    @classmethod
+    def create_missing(cls) -> None:
```

Comment:
Somewhat contrary to what I commented earlier, perhaps using a transaction in the creation stage makes sense to avoid potential situations where an aggregate isn't associated with any community? I'd go as far as saying it should probably lock the communities too to avoid race conditions (check `.select_for_update` queryset docs).

The update method still shouldn't require a transaction however, it doesn't really matter even if it ends up overlapping with itself.

Reply:
Added

Reply:
Interesting, but makes sense 👍 I think we can leave it without a lock since a race condition should basically never happen, and even if it did it would be a very minor problem.